akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/ClusterShardingSettings.scala

```diff
@@ -265,15 +283,36 @@ final class ClusterShardingSettings(
     val passivateIdleEntityAfter: FiniteDuration,
     val shardRegionQueryTimeout: FiniteDuration,
     val stateStoreMode: ClusterShardingSettings.StateStoreMode,
+    val rememberEntitiesStoreMode: ClusterShardingSettings.RememberEntitiesStoreMode,
     val tuningParameters: ClusterShardingSettings.TuningParameters,
     val coordinatorSingletonSettings: ClusterSingletonManagerSettings) {
 
-  import akka.cluster.sharding.typed.ClusterShardingSettings.StateStoreModeDData
-  import akka.cluster.sharding.typed.ClusterShardingSettings.StateStoreModePersistence
-  require(
-    stateStoreMode == StateStoreModePersistence || stateStoreMode == StateStoreModeDData,
-    s"Unknown 'state-store-mode' [$stateStoreMode], " +
-    s"valid values are '${StateStoreModeDData.name}' or '${StateStoreModePersistence.name}'")
+  @deprecated("Use constructor with rememberEntitiesStoreMOde", "2.6.6") // FIXME update version once merged
+  def this(
+      numberOfShards: Int,
+      role: Option[String],
+      dataCenter: Option[DataCenter],
+      rememberEntities: Boolean,
+      journalPluginId: String,
+      snapshotPluginId: String,
+      passivateIdleEntityAfter: FiniteDuration,
+      shardRegionQueryTimeout: FiniteDuration,
+      stateStoreMode: ClusterShardingSettings.StateStoreMode,
+      tuningParameters: ClusterShardingSettings.TuningParameters,
+      coordinatorSingletonSettings: ClusterSingletonManagerSettings) =
+    this(
+      numberOfShards,
+      role,
+      dataCenter,
+      rememberEntities,
+      journalPluginId,
+      snapshotPluginId,
+      passivateIdleEntityAfter,
+      shardRegionQueryTimeout,
+      stateStoreMode,
+      RememberEntitiesStoreModeDData,
+      tuningParameters,
+      coordinatorSingletonSettings)
```

Comment:
Do'h I pretty much just spent time on the exact same change adding it in settings and a small ADT for store modes 🤦 

I'll try to communicate better what I'm up to moving forward.